### PR TITLE
Make artillery rotate to target before firing

### LINKED ISSUE #2775: [Bug]: Artillery possible unable to fire
### Describe the bug

Some artillery units turrets, e.g. MLRS systems from GM don't have a 360° rotation on the turret/launcher complex

### How to reproduce

Go ingame test mlrs.

### Version

3.x

### Have you altered the code?

No

### What i have changed

_No response_

### Map

every

### What server?

LAN Hosted

### Time bug occured (Server time/UTC)

_No response_

### Mods

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jaj22**
### What type of PR is this.
1. [X] Bug
2. [X] Change
3. [ ] Enhancement

### What have you changed and why?
Some artillery vehicles have a limited firing arc and the artillery support code didn't take account of that. This PR fixes the issue by making artillery vehicles rotate towards the target before firing. Method should be solid. Adjusted the firing rate to 1/sec max as previously it varied based on rotation distance.

Not tested with anything except vanilla mortars, but should work fine.

### Please specify which Issue this PR Resolves.
closes #2775

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Adjust as required:
`["MORTAR", Occupants, "defence", 500, objNull, markerPos "Therisa", 0.8, 20] call A3A_fnc_createSupport;`

### CODE REVIEW

File: A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf [hunk 1]
```diff
@@ -67,6 +67,27 @@ _fn_executeMortarFire =
...
+{
+    params ["_mortar", "_targPos"];
+
+    private _change = (_mortar getDir _targPos) - getDir _mortar;
+    _change = (_change + 540) % 360 - 180;
+
+    addMissionEventHandler ["EachFrame", {
+        _thisArgs params ["_mortar", "_startDir", "_change", "_startTime"];
+
+        private _interval = 10 * (time - _startTime) / abs _change;        // 10 degree/sec turn
```
Comment (ante185) [inline A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf:81]: possible zero divisor

**Review by ante185 — CHANGES_REQUESTED** [review]

File: A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf (hunk 1 above)
Comment (jaj22) [reply A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf:81]: Good call.

File: A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf [hunk 2]
```diff
@@ -55,7 +55,7 @@ _fn_executeMortarFire =
             [_shellTarget, _mortar] spawn
             {
                 params ["_shellTarget", "_mortar"];
-                sleep 0.5;
+                sleep 1;
```
Comment (Tiny-DM) [inline A3A/addons/core/functions/Supports/fn_SUP_mortarRoutine.sqf:58]: What is this change for? Just seems to lengthen the delay between the request and the firing

**Review by ante185 — APPROVED** [review]
Tested with IFA as this modset offers the artillery that'd require the ability to rotate.
For mortars it looks just like the regular aiming rotation, so that's about as good as is possible.
Bigger pieces do look odd, but given the low probability of seeing this occurring this is a concession that's gonna be alright to make.

TLDR: lgtm

**Review by Tiny-DM — APPROVED** [review]
lgtm
